scalding-core/src/main/scala/com/twitter/scalding/Execution.scala

```diff
@@ -192,6 +193,8 @@ sealed trait Execution[+T] extends java.io.Serializable {
    */
   def zip[U](that: Execution[U]): Execution[(T, U)] =
     Zipped(this, that)
+
+  override val hashCode = super.hashCode
```

Comment:
so, `Mapped, FlatMapped, Zipped, OnComplete` would have strict `hashCode` that calls `p, but the others would not need it? We could just call this: https://github.com/scala/scala/blob/2.12.x/src/library/scala/runtime/ScalaRunTime.scala#L145 and store it in a val?